examples/js/renderers/SVGRenderer.js

```diff
@@ -87,6 +87,27 @@ THREE.SVGRenderer = function () {
 
 	this.setPixelRatio = function () {};
 
+	this.getSize = function () {
+
+		var width = _svgWidth;
+		var height = _svgHeight;
+		if ( width === undefined || height === undefined ) {
+
+			// If setSize has not been called, we use the bounding client rect,
+			// rather than the SVG contents rect (getBBox), to respect CSS based sizing.
+			let box = _svg.getBoundingClientRect();
+			width = width === undefined ? box.width : width;
+			height = height === undefined ? box.height : height;
+
+		}
+
+	 	return { 
```

Comment:
It should be sufficient to just return `_svgWidth` and `_svgHeight`. Please keep the implementation simple.

Reply:
Unlike WebGLRenderer that defaults values for these, SVGRenderer leaves them undefined. This means that a call to getSize that has never called setSize gets { width: undefined, height: undefined }... which seems unexpected?

Reply:
Why should `setSize()` be never called? The usual pattern in all examples is to create the renderer and then set its size.

Reply:
When render.domElement.parent controls the size of render.domElement through CSS, setSize might not be called until you actually need to update the camera's projection matrix in an animate step. Meanwhile, you might call getSize prior to that while setting up your camera or adjusting zoom or the frustrum.

In the case of WebGLRenderer, `_width` and `_height` default to `_canvas.width` and `_canvas.height`, respectively. In SVGRenderer, `_svgWidth` and `_svgHeight` are left undefined. HTML5 specifies that a canvas defaults to 300x150, and it does the same for svg elements, but the some browsers expand inline svg elements to fill the viewport (`width: 100vw; height: 100vh;`). Either way, the svg element will have a width and a height and I can change the `THREE.SVGRenderer` constructor to set:
```javascript
  _svgWidth = _svg.baseVal.width;
  _svgHeight = _svg.baseVal.height;
```
If that would be preferred over the getSize implementation I used, but simply leaving getSize() to return undefined values seems wrong.

Reply:
Actually, I take it back. The svg element is not yet connected to the dom, so `_svg.width.baseVal.value` can't be resolved in Chromium browsers, as Chromium defaults to 100%. I could do:
```javascript
	this.getSize = function () {

		if ( ( _svgWidth === undefined || _svgHeight === undefined ) && _svg.parentElement === undefined ) {

			console.warn( "SVGRenderer: .getsize() can't be called until after setSize or after .domElement has been attached to the dom" );

		}

		return { 
			width: _svgWidth ? _svgWidth : _svg.width.baseVal.value, 
			height: _svgHeight ? _svgHeight : _svg.height.baseVal.value
		};

	};
```

Reply:
Sorry but you make the code unnecessarily complicated. I can only approve this version:
```js
this.getSize = function () {

	return {
		width: _svgWidth,
		height: _svgHeight
	};

};
```

Reply:
I don't understand how providing a concise warning when developers misuse an API makes the code "unnecessarily complicated"? There are plenty of examples of users that "fiddle around with CSS in order to size the renderer's canvas", including:

- https://threejsfundamentals.org/threejs/lessons/threejs-responsive.html
- https://github.com/mrdoob/three.js/issues/4903
- https://stackoverflow.com/questions/49344698/threejs-size-canvas-to-fill-space

and support for this was added to WebGLRenderer back in 2013: https://github.com/mrdoob/three.js/commit/ed00cd63fac6a13c55ccdd40c7c55d6931c118de (and expanded in 2016: https://github.com/mrdoob/three.js/commit/5f78a2786b0a3231501ec0d6d429096ac3419a54 ) in response to https://github.com/mrdoob/three.js/issues/2969

Reply:
The mentioned guide `Three.js Responsive Design` and #4903 are from the same dev and I don't share his opinion on this topic. I think it's better if the engine does not require the user to work with CSS.